tests/subworkflows/install.py

```diff
@@ -0,0 +1,78 @@
+
+from nf_core.modules.modules_json import ModulesJson
+from nf_core.subworkflows.install import SubworkflowInstall
+
+from ..utils import (
+    GITLAB_BRANCH_TEST_BRANCH,
+    GITLAB_REPO,
+    GITLAB_SUBWORKFLOWS_BRANCH,
+    GITLAB_URL,
+    with_temporary_folder,
+)
+
+
+def test_subworkflow_install_nopipeline(self):
+    """Test installing a subworkflow - no pipeline given"""
+    self.subworkflow_install.dir = None
+    assert self.subworkflow_install.install("foo") is False
+
+
+@with_temporary_folder
+def test_subworkflows_install_emptypipeline(self, tmpdir):
+    """Test installing a subworkflow - empty dir given"""
+    os.mkdir(os.path.join(tmpdir, "nf-core-pipe"))
+    self.subworkflow_install.dir = os.path.join(tmpdir, "nf-core-pipe")
+    with pytest.raises(UserWarning) as excinfo:
+        self.subworkflow_install.install("foo")
+    assert "Could not find a 'main.nf' or 'nextflow.config' file" in str(excinfo.value)
+
+
+def test_subworkflows_install_nosubworkflow(self):
+    """Test installing a subworkflow - unrecognised subworkflow given"""
+    assert self.subworkflow_install.install("foo") is False
+
+
+def test_subworkflows_install_bam_sort_stats_samtools(self):
+    """Test installing a subworkflow - bam_sort_stats_samtools"""
+    assert self.subworkflow_install.install("bam_sort_stats_samtools") is not False
+    subworkflow_path = os.path.join(self.subworkflow_install.dir, "subworkflows", "nf-core", "bam_sort_stats_samtools")
+    sub_subworkflow_path = os.path.join(self.subworkflow_install.dir, "subworkflows", "nf-core", "bam_stats_samtools")
+    samtools_index_path = os.path.join(self.subworkflow_install.dir, "modules", "nf-core", "samtools", "index")
+    samtools_sort_path = os.path.join(self.subworkflow_install.dir, "modules", "nf-core", "samtools", "sort")
+    samtools_stats_path = os.path.join(self.subworkflow_install.dir, "modules", "nf-core", "samtools", "stats")
+    samtools_idxstats_path = os.path.join(self.subworkflow_install.dir, "modules", "nf-core", "samtools", "idxstats")
+    samtools_flagstat_path = os.path.join(self.subworkflow_install.dir, "modules", "nf-core", "samtools", "flagstat")
+    assert os.path.exists(subworkflow_path)
+    assert os.path.exists(sub_subworkflow_path)
+    assert os.path.exists(samtools_index_path)
+    assert os.path.exists(samtools_sort_path)
+    assert os.path.exists(samtools_stats_path)
+    assert os.path.exists(samtools_idxstats_path)
+    assert os.path.exists(samtools_flagstat_path)
+
+
+def test_subworkflows_install_bam_sort_stats_samtools_twice(self):
+    """Test installing a subworkflow - bam_sort_stats_samtools already there"""
+    self.subworkflow_install.install("bam_sort_stats_samtools")
+    assert self.subworkflow_install.install("bam_sort_stats_samtools") is False
+
+
+def test_subworkflows_install_from_gitlab(self):
+    """Test installing a subworkflow from GitLab"""
+    assert self.subworkflow_install_gitlab.install("bam_stats_samtools") is True
```

Comment:
I ran also into this with the list tests. Should we update the repo there, so we can use the same subworkflow tool, to avoid confusion? Or is this actually making it a bit more fool proof?

Reply:
We can update the repo, I was being lazy, as it's only the name that changes